model/task_lifecycle.go

```diff
@@ -817,6 +817,11 @@ func UpdateUnblockedDependencies(t *task.Task) error {
 		if err := UpdateUnblockedDependencies(&blockedTask); err != nil {
 			return errors.WithStack(err)
 		}
+
+		if err := UpdateBuildAndVersionStatusForTask(&blockedTask); err != nil {
```

Comment:
Rather than updating the build/version for every single iteration, can this do the same status update logic as in `UpdateBlockedDependencies` and collect a list of build IDs to update at the end? It would avoid having to update the statuses repeatedly for the same builds.

Reply:
done